src/main.rs

```diff
@@ -17,14 +17,22 @@ async fn main() -> eyre::Result<()> {
 
     let tasks = vec![save_run(core.clone(), out_dir, SsuCatchPlugin::default())];
 
+    let mut is_successful = false;
     join_all(tasks).await.into_iter().for_each(|result| {
         if let Err(err) = result {
             eprintln!("{:?}", err);
+        } else {
+            is_successful = true;
         }
     });
 
     core.save_cache().await?;
-    Ok(())
+
+    if is_successful {
+        Ok(())
+    } else {
+        panic!("all plugin failed")
```

Comment:
p3) 하나라도 실패하더라도 exit code 1로 나가야 하지 않을까요? `for_each`로 에러를 그냥 소비하지 않고, map 등으로 aggregate 해서 expect나 Err를 리턴하여 main 함수를 종료하는 식으로 끝내면 더 좋을 것 같습니다.

Reply:
일부 플러그인에서 크롤링이 실패하더라도 크롤링이 성공한 플러그인의 결과는 반영되도록 하는게 좋지 않을까요?

명시적으로 `panic!`을 호출하는 것보다는 `Iterator`의 메서드를 활용해서 더 가독성 좋게 개선할 수 있을 것 같아요! (예) `filter()`로 `Ok`를 반환한 Result만 모아서 배열의 길이가 0일 때만 `Err`를 반환)

Reply:
> @2wndrhs 일부 플러그인에서 크롤링이 실패하더라도 크롤링이 성공한 플러그인의 결과는 반영되도록 하는게 좋지 않을까요?

이 부분은 동의해요. 하지만 전체 크롤링이 끝난 뒤에 하나라도 실패한 플러그인이 있으면 exit code가 1이 되어야 한다는 의미였어요(중간에 강종 X). 제가 헷갈리게 적었나 보네요 ㅎㅎ;;

Reply:
> filter()로 Ok를 반환한 Result만 모아서 배열의 길이가 0일 때만 Err를 반환)

filter로 `Err`만 모으면 에러 정보와 실패 여부를 한번에 알 수 있어서 좋을 것 같네요.

Reply:
> 하지만 전체 크롤링이 끝난 뒤에 하나라도 실패한 플러그인이 있으면 exit code가 1이 되어야 한다는 의미였어요(중간에 강종 X)

exit code가 1이면 워크플로에서 다음 스텝을 수행하지 않는 걸로 아는데 맞을까요?
플러그인의 결과가 반영되려면 다음 스텝에 있는 deploy-pages를 수행해야 해서 모두 실패한 게 아니라면 exit code는 0이어야 할 것 같습니다!

Reply:
> exit code가 1이면 워크플로에서 다음 스텝을 수행하지 않는 걸로 아는데 맞을까요?
워크플로에서 `if: always()` 단 step은 결과와 별개로 수행하는 기능이 있습니다. 워크플로와는 별개로 생각했을 때 하나라도 실패 시 exit code 가 1이 되는 동작은 정상이라고 생각해요.

그래서 데몬 개발 시에는 워크플로를 생각하지 않고, 워크플로가 데몬의 동작에 맞추어 작성되어야 하지 않을까 싶네요.

Reply:
Err만 모은 벡터가 비어있지 않은 경우 exit code 1으로 종료하도록 수정했습니다
https://github.com/yourssu/ssufid/pull/59/commits/dbd86ee22fda7d31a37335f7647f4b59c08fb6a3

Reply:
에러를 반환하지 않고 `exit(1)`을 호출하여 종료하는 이유가 있을까요?

`exit()`함수는 소멸자 호출이 이루어지지 않을 수 있다고 해서요.

> Note that because this function never returns, and that it terminates the process, no destructors on the current stack or any other thread’s stack will be run.
> https://doc.rust-lang.org/std/process/fn.exit.html

```rust
return Err(eyre::eyre!("Some tasks failed"));
```

Reply:
아 `eyre::Result`를 반환하는 함수에서 명시적으로 Err 반환하는 방법을 몰라서 exit 한거였어요🥲 
말씀해주신 방법대로 수정하는 게 좋겠네요..!